File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/visitor/hosting/FileStorageService.java
package com.bigboxcode.designpattern.visitor.hosting;

public class FileStorageService implements Service {

    private final double pricePerGB = 1.70;
    private int quantity;

    public FileStorageService(int quantity) {
        this.quantity = quantity;
    }

    public double getPricePerGB() {
        return pricePerGB;
    }

    public int getQuantity() {
        return quantity;
    }

    @Override
    public double accept(HostingCalculatorVisitor hostingCalculatorVisitor) {
        return hostingCalculatorVisitor.visit(this);
    }
}

File: design-patterns/php/visitor/hostingcost/FileStorageService.php
<?php
// FileStorageService.php

namespace BigBoxCode\DesignPattern\Visitor\HostingCost;


class FileStorageService implements Service {
    private const PRICE_PER_GB = 1.70;
    private int $quantity;

    public function __construct(int $quantity) {
        $this->quantity = $quantity;
    }

    public function getPricePerGB(): float {
        return self::PRICE_PER_GB;
    }

    public function getQuantity(): int {
        return $this->quantity;
    }

    public function accept(HostingCalculatorVisitor $hostingCalculatorVisitor): float {
        return $hostingCalculatorVisitor->visitFileStorageService($this);
    }
}